InformationScripting/src/queries/CompositeQuery.cpp

```diff
@@ -76,7 +77,13 @@ QList<TupleSet> CompositeQuery::execute(QList<TupleSet> input)
 					return m.outputFrom_ == currentNode && m.outputIndex_ == outIndex;
 				});
 				Q_ASSERT(inputIt != receiver->inputMap_.end());
-				auto output = hasOutput ? currentNode->calculatedOutputs_[outIndex] : TupleSet();
+				TupleSet output;
+				if (hasOutput)
+				{
+					// early abort in case of error:
+					if (currentNode->calculatedOutputs_[outIndex]) output = currentNode->calculatedOutputs_[outIndex].value();
+					else return { currentNode->calculatedOutputs_[outIndex].error() };
```

Comment:
About the first comment, in that place there should only ever be one error (otherwise we would have aborted already), but maybe multiple warnings.

I will have to think about Optional<QList vs QList<Optional